intel/custom_resource.py

```diff
@@ -0,0 +1,213 @@
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import util
+from kubernetes.client.rest import ApiException as K8sApiException
+import datetime
+import json
+import logging
+import time
+
+
+# Example usage:
+#
+# from kubernetes import config as k8sconfig, client as k8sclient
+#
+# k8sconfig.load_kube_config()
+#
+# v1beta = k8sclient.ExtensionsV1beta1Api()
+#
+# apple_type = CustomResourceDefinitionType(
+#               v1beta, "intel.com", "apple", ["ap"])
+#
+# apple1 = apple_type.create("apple1")
+# apple1.body["spec"]["count"] = 5
+# apple1.save()
+#
+# apple2 = apple_type.create("apple2")
+# apple2.body["spec"]["count"] = 10
+# apple2.save()
+
+
+class CustomResourceDefinitionType():
+    def __init__(self, api, group, name, short_names,
+                 version="v1", scope="Namespaced"):
+        assert (api is not None)
+        assert (group is not None)
+        assert (name is not None)
+        assert (short_names is not None)
+        assert (version is not None)
+        assert (scope is not None)
+
+        self.api = api
+        self.group = group
+        self.kind_name = str(name).capitalize()
+        self.singular_name = name
+        self.plural_name = "{}s".format(name)
+        self.short_names = short_names
+        self.type_version = version
+        self.type_scope = scope
+
+    def save(self):
+
```

Comment:
Public methods should have docstrings.
(https://www.python.org/dev/peps/pep-0257/)

Replace this empty newline with the docstring